scipy/io/matlab/tests/test_mio.py

```diff
@@ -553,24 +552,12 @@ def test_use_small_element():
 
 def test_save_dict():
     # Test that dict can be saved (as recarray), loaded as matstruct
-    dict_types = ((dict, False), (OrderedDict, True),)
     ab_exp = np.array([[(1, 2)]], dtype=[('a', object), ('b', object)])
-    ba_exp = np.array([[(2, 1)]], dtype=[('b', object), ('a', object)])
-    for dict_type, is_ordered in dict_types:
-        # Initialize with tuples to keep order for OrderedDict
-        d = dict_type([('a', 1), ('b', 2)])
-        stream = BytesIO()
-        savemat(stream, {'dict': d})
-        stream.seek(0)
-        vals = loadmat(stream)['dict']
-        assert_equal(set(vals.dtype.names), set(['a', 'b']))
-        if is_ordered:  # Input was ordered, output in ab order
-            assert_array_equal(vals, ab_exp)
-        else:  # Not ordered input, either order output
-            if vals.dtype.names[0] == 'a':
-                assert_array_equal(vals, ab_exp)
-            else:
-                assert_array_equal(vals, ba_exp)
+    stream = BytesIO()
+    savemat(stream, {'dict': {'a': 1, 'b': 2}})
+    stream.seek(0)
+    vals = loadmat(stream)['dict']
+    assert_array_equal(vals, ab_exp)
```

Comment:
Thanks for that catch. Testing `savemat` using `OrderedDict` has been restored, but this test has otherwise been simplified to assume the items are ordered.